### PR TITLE
[@types/react] add template to IntrinsicElements

### PR DESCRIPTION
**Pull request by magic-akari**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by magic-akari** [comment]
> The job exceeded the maximum time limit for jobs, and has been terminated.
It looks like there is no error. But the CI runs out of time. 
What should I do?

File: types/react/index.d.ts [hunk 1]
```diff
@@ -2593,6 +2593,7 @@ declare namespace React {
         summary: DetailedHTMLFactory<HTMLAttributes<HTMLElement>, HTMLElement>;
         sup: DetailedHTMLFactory<HTMLAttributes<HTMLElement>, HTMLElement>;
         table: DetailedHTMLFactory<TableHTMLAttributes<HTMLTableElement>, HTMLTableElement>;
+        template: DetailedHTMLFactory<HTMLAttributes<HTMLElement>, HTMLElement>;
```
Comment (eps1lon) [inline types/react/index.d.ts]: Looks like [`templaate`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) even implements [its own interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTemplateElement). Is this available in the dom lib?

**Review by eps1lon — COMMENTED** [review]
Never used this element. Is it fully implemented in `react-dom`?

File: types/react/index.d.ts (hunk 1 above)
Comment (magic-akari) [reply types/react/index.d.ts]: > Looks like [`templaate`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) even implements [its own interface](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTemplateElement). Is this available in the dom lib?

Yes. It's my fault to forget this. I'll correct it.

**Pull request closed without merging**